tests/integration/api/test_vector_aggregation_operations.py

```diff
@@ -86,37 +159,193 @@ async def register_table_with_array_column(
+        item_id_column="ITEM_ID",
+        event_table_name=event_table_with_array_column.name,
+    )
+    item_table["ITEM_ID"].as_entity(item_entity.name)
+    item_table["ORDER_ID"].as_entity(vector_order_entity.name)
+    return item_table
+
+
+@pytest_asyncio.fixture(name="scd_table_with_array_column", scope="module")
+async def register_scd_table_with_array_column(
+    scd_data_with_array, session, data_source, catalog, vector_user_entity
+):
+    """
+    Register a table with an array column
+    """
+    table_name = "scd_table_with_vector"
+    await session.register_table(table_name, scd_data_with_array, temporary=False)
+
+    database_table = data_source.get_source_table(
+        database_name=session.database_name,
+        schema_name=session.schema_name,
+        table_name=table_name,
+    )
+    scd_table = database_table.create_scd_table(
+        name=table_name,
+        natural_key_column="ITEM_ID",
+        effective_timestamp_column="EVENT_TIMESTAMP",
+    )
+    scd_table["USER_ID"].as_entity(vector_user_entity.name)
+    return scd_table
+
+
+TEST_CASES = [
+    (AggFunc.MAX, [3.0, 3.0, 3.0], VECTOR_VALUE_FLOAT_COL),
+    (AggFunc.AVG, [2.0, 2.0, 2.0], VECTOR_VALUE_FLOAT_COL),
+    (AggFunc.SUM, [4.0, 4.0, 4.0], VECTOR_VALUE_FLOAT_COL),
+    (AggFunc.MAX, [3.0, 3.0, 3.0], VECTOR_VALUE_MIXED_COL),
+    (AggFunc.AVG, [2.0, 2.0, 2.0], VECTOR_VALUE_MIXED_COL),
+    (AggFunc.SUM, [4.0, 4.0, 4.0], VECTOR_VALUE_MIXED_COL),
+    (AggFunc.MAX, [3.0, 3.0, 3.0], VECTOR_VALUE_INT_COL),
+    (AggFunc.AVG, [2.0, 2.0, 2.0], VECTOR_VALUE_INT_COL),
+    (AggFunc.SUM, [4.0, 4.0, 4.0], VECTOR_VALUE_INT_COL),
+]
+
+
 @pytest.mark.parametrize("source_type", ["spark"], indirect=True)
-def test_vector_aggregation_operations(event_table_with_array_column):
+@pytest.mark.parametrize(
+    "agg_func,expected_results,vector_value_column",
+    TEST_CASES,
+)
+def test_vector_aggregation_operations__aggregate_over(
+    event_table_with_array_column, agg_func, expected_results, vector_value_column
+):
     """
     Test vector aggregation operations
     """
     event_view = event_table_with_array_column.get_view()
-    feature = event_view.groupby("CUST_ID").aggregate_over(
-        value_column="VECTOR_VALUE",
-        method=AggFunc.MAX,
+    feature_name = "vector_agg"
+    feature = event_view.groupby("USER_ID").aggregate_over(
+        value_column=vector_value_column,
+        method=agg_func,
         windows=["1d"],
-        feature_names=["vector_max"],
+        feature_names=[feature_name],
+        skip_fill_na=True,
+    )[feature_name]
+
+    preview_params = {"POINT_IN_TIME": "2022-06-06 00:58:00", "vector_user_id": "2"}
+    feature_preview = feature.preview(pd.DataFrame([preview_params]))
+    assert feature_preview.shape[0] == 1
+    assert feature_preview.iloc[0].to_dict() == {
+        feature_name: expected_results,
+        **convert_preview_param_dict_to_feature_preview_resp(preview_params),
+    }
```

Comment:
Would be good to also test computing historical features.

Reply:
sounds good, will update too.